### PR TITLE
AuthorMappingPane: use `useUsersQuery` to fetch total users

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

* `<AuthorMappingPane />` fetch total by using `useUsersQuery` instead of `<SiteUsersFetcher />`

#### Testing instructions

* On a site with >1 author start importing content to your site until you reach the author mapping step
* Make sure the mapping description is displayed correctly (see screenshot)

Screenshot 2021-04-13 at 10 43 45

Related to #24150

### CODE REVIEW

File: client/my-sites/importer/author-mapping-pane.jsx [hunk 1]
```diff
@@ -181,4 +171,13 @@ class AuthorMappingPane extends React.PureComponent {
 	}
 }
 
-export default localize( AuthorMappingPane );
+const withTotalUsers = ( Component ) => ( props ) => {
```
Comment (tyxla) [inline client/my-sites/importer/author-mapping-pane.jsx:174]: Have you considered using `createHigherOrderComponent` here? It nicely takes care of the name that's displayed for the higher-order component when rendered in the dev tools.

File: client/my-sites/importer/author-mapping-pane.jsx [hunk 2]
```diff
@@ -181,4 +171,13 @@ class AuthorMappingPane extends React.PureComponent {
...
+	const { siteId } = props;
+	const { data } = useUsersQuery( siteId, {}, { refetchOnWindowFocus: false } );
```
Comment (tyxla) [inline client/my-sites/importer/author-mapping-pane.jsx:176]: Is there a strong reason to disable `refetchOnWindowFocus`? Are we trying to prevent extra rerenders or something?

Comment (tyxla) [inline client/my-sites/importer/author-mapping-pane.jsx:176]: In the previous version of the code, we were specifically fetching up to 50 users. With the current default of `useUsersQuery`, we'll now be fetching 100. Is that change intentional?

**Review by tyxla — COMMENTED** [review]
This works great in my testing 👍 

I've just left some questions, let me know what you think.

File: client/my-sites/importer/author-mapping-pane.jsx (hunk 2 above)
Comment (flootr) [reply client/my-sites/importer/author-mapping-pane.jsx:176]: I opted for default params so it would reuse existing queries cache; `<AuthorSelector />` is used on the same page and also uses `useUsersQuery` now. Although `number` isn't used to construct the `cacheKey` so that assumption might be wrong. Nevertheless, there's no need to fetch more than 1 user as we're only interested in `total`.

Good call though, I've to verify that behaviour.

Comment (flootr) [reply client/my-sites/importer/author-mapping-pane.jsx:176]: Hm… I wanted to prevent overfetching. However, we can do that by using the `staleTime` param as well. I've gotta give this more thought.

File: client/my-sites/importer/author-mapping-pane.jsx (hunk 1 above)
Comment (flootr) [reply client/my-sites/importer/author-mapping-pane.jsx:174]: Good call! I'll try that 👍🏻

File: client/my-sites/importer/author-mapping-pane.jsx (hunk 2 above)
Comment (tyxla) [reply client/my-sites/importer/author-mapping-pane.jsx:176]: >  Nevertheless, there's no need to fetch more than 1 user as we're only interested in total.

That sounds interesting, and my expectation is that it would make the API response even faster, so it sounds like an interesting option! Let me know how it goes once you try it.

Comment (tyxla) [reply client/my-sites/importer/author-mapping-pane.jsx:176]: If we leave it as it is by default, it would poll for users at a certain interval, right? That could help keep the freshness of the component - if you delete a user in another tab, it would keep the data in this one fresh in case it alters the logic based on the number of users. Maybe that's not a bad thing to have? It's by default, too.

**Review by tyxla — APPROVED** [review]
FWIW, this looks good, and I think we can merge it and follow-up with the potential improvements - none of them are blockers. 😉
